src/Refactor/Entity/Product.php

```diff
@@ -0,0 +1,222 @@
+<?php
+namespace DemandwareXml\Refactor\Entity;
+
+use DateTimeInterface;
+use DemandwareXml\Refactor\EntityWriter\ProductXmlWriter;
+use DemandwareXml\Refactor\Xml\XmlWriter;
+use InvalidArgumentException;
+
+class Product implements WriteableEntityInteface
+{
+    public $id;
+    public $upc;
+    public $minOrderQuantity;
+    public $stepQuantity;
+    public $displayName;
+    public $longDescription;
+    public $onlineFlag;
+    public $onlineFrom;
+    public $onlineTo;
+    public $availableFlag;
+    public $searchableFlag;
+    public $searchableIfUnavailableFlag;
+    public $tax;
+    public $images                    = [];
+    public $imageViewType;
+    public $brand;
+    public $searchRank;
+    public $sitemapIncludedFlag;
+    public $sitemapChangeFrequency;
+    public $sitemapPriority;
+    public $pageAttributes            = [];
+    public $customAttributes          = [];
+    public $sharedVariationAttributes = [];
+    public $variants                  = [];
+    public $bundleProducts            = null;
+    public $setProducts               = null;
+    public $classificationCategoryId;
+    public $classificationCatalogId;
+
+    public function __construct(string $id)
+    {
+        $this->id = $id;
+    }
+
+    public function setUpc(string $upc): void
+    {
+        $this->upc = $upc;
+    }
+
+    public function setQuantities(int $minOrderQuantity = 1, int $stepQuantity = 1): void
+    {
+        $this->minOrderQuantity = $minOrderQuantity;
+        $this->stepQuantity     = $stepQuantity;
+    }
+
+    public function setDisplayName(string $displayName): void
+    {
+        $this->displayName = $displayName;
+    }
+
+    public function setLongDescription(string $longDescription): void
+    {
+        $this->longDescription = $longDescription;
+    }
+
+    public function setOnlineFlag(bool $onlineFlag): void
+    {
+        $this->onlineFlag = $onlineFlag;
+    }
+
+    public function setOnlineFromTo(?DateTimeInterface $from, ?DateTimeInterface $to): void
+    {
+        if ($from !== null) {
+            $this->onlineFrom = $from;
+        }
+
+        if ($to !== null) {
+            $this->onlineTo = $to;
+        }
+    }
+
+    public function setSearchableFlags(?bool $availableFlag = null, ?bool $searchableFlag = null, ?bool $searchableIfUnavailableFlag = null): void
```

Comment:
I think we should use either nullable type (`?`) notation _or_ a default `null` value, but we don't need both. The only difference between the two is that the `null` default allows us to pass nothing for that argument, which will then fallback to its own `null` default.

Reply:
@WillJW I've changed this here and on the `setPageAttributes` function too.